omegaconf/basecontainer.py

```diff
@@ -224,9 +224,37 @@ def pretty(self, resolve: bool = False, sort_keys: bool = False) -> str:
         """
         from omegaconf import OmegaConf
 
+        dumper = yaml.Dumper
+
+        def str_representer(dumper: yaml.Dumper, data: str) -> yaml.ScalarNode:
+            def is_bool(data: str) -> bool:
+                return data in ["True", "False"]
+
+            def is_int(data: str) -> bool:
+                return data.isdecimal()
+
+            def is_float(data: str) -> bool:
+                is_float = True
+                try:
+                    float(data)
+                except ValueError:
+                    is_float = False
+                return is_float
+
+            with_quotes = is_bool(data) or is_int(data) or is_float(data)
+            return dumper.represent_scalar(
+                yaml.resolver.BaseResolver.DEFAULT_SCALAR_TAG,
+                data,
+                style=("single-quoted" if with_quotes else None),
+            )
+
+        dumper.add_representer(str, str_representer)
```

Comment:
This is the method that changes the representer of a certain datatype.

Reply:
This method is changing the actual pyyaml dumper.
You already fixed on bug caused by doing something similar :)

define your own dumper (in _utils) and use it here with

```python
def dump(data, stream=None, Dumper=Dumper, **kwds):
```